app/src/colorpalettewidget.cpp

```diff
@@ -262,6 +264,54 @@ void ColorPaletteWidget::onItemChanged(QListWidgetItem* item)
     editor()->object()->renameColour(index, newColorName);
 }
 
+void ColorPaletteWidget::onRowsMoved(const QModelIndex &parent, int start, int end, const QModelIndex &destination, int row)
+{
+    Q_UNUSED(parent)
+    Q_UNUSED(destination)
+    Q_UNUSED(end)
+
+    int startIndex, endIndex;
+    if (start < row)
+    {
+        row -= 1; // TODO: Is this a bug?
+        if (start == row) { return; }
+
+        startIndex = start;
+        endIndex = row;
+
+        editor()->object()->movePaletteColor(startIndex, endIndex);
+
+        editor()->object()->addColour(editor()->object()->getColour(startIndex));
+        editor()->object()->moveVectorColor(startIndex, editor()->object()->getColourCount() - 1);
+        for (int i = startIndex; i < endIndex; i++)
+        {
+            editor()->object()->moveVectorColor(i + 1, i);
+        }
+        editor()->object()->moveVectorColor(editor()->object()->getColourCount() - 1, endIndex);
+    }
+    else
+    {
+        if (start == row) { return; }
+
+        startIndex = start;
+        endIndex = row;
+
+        editor()->object()->movePaletteColor(startIndex, endIndex);
+
+        editor()->object()->addColour(editor()->object()->getColour(startIndex));
+        editor()->object()->moveVectorColor(startIndex, editor()->object()->getColourCount() - 1);
+        for (int i = startIndex; i > endIndex; i--)
+        {
+            editor()->object()->moveVectorColor(i - 1, i);
+        }
+        editor()->object()->moveVectorColor(editor()->object()->getColourCount() - 1, endIndex);
+    }
+
+    editor()->object()->removeColour(editor()->object()->getColourCount() - 1);
```

Comment:
You call editor()->object a lot, consider creating a local Object* variable so that repeated reference lookup is avoided.

Reply:
For some odd reason I had to make both Editor and Object local vars.
When I tried
`mObject = editor()->object();`
it crashed. It works fine now, with the usual setCore method.